sdk/rust/oak/src/storage/mod.rs

```diff
@@ -45,104 +43,30 @@ impl Storage {
     /// Create a `Storage` instance using the given name identifying storage
     /// node configuration.
     pub fn new(config: &str) -> Option<Storage> {
-        // Create a channel and pass the read half to a fresh storage Node.
-        let (write_channel, read_channel) = match crate::channel_create() {
-            Ok(x) => x,
-            Err(_e) => return None,
-        };
-        if crate::node_create(config, "oak_main", read_channel).is_err() {
-            if let Err(status) = crate::channel_close(write_channel.handle) {
-                warn!("could not close write channel: {:?}", status);
-            };
-            if let Err(status) = crate::channel_close(read_channel.handle) {
-                warn!("could not close read channel: {:?}", status);
-            };
-            return None;
-        }
-
-        if let Err(status) = crate::channel_close(read_channel.handle) {
-            warn!("could not close read channel: {:?}", status);
-        };
-        Some(Storage { write_channel })
+        crate::grpc::client::Client::new(config, "oak_main").map(|client| Storage { client })
     }
 
     fn execute_operation<Req, Res>(
         &mut self,
         grpc_method_name: &str,
-        operation_request: &Req,
+        operation_request: Req,
```

Comment:
It seems a bit too strict to require this to be passed by value, even if passed by reference and not used afterwards I am guessing the compiler would optimize it away.

edit: it seems the requirement stems from the line below, but I think even that may be changed into a reference?

https://github.com/project-oak/oak/blob/a39e166e77ebcdf79caaa34ef379deadb91aabcd/sdk/rust/oak/src/grpc/mod.rs#L171